src/Tests/IQsharpEngineTests.cs

```diff
@@ -938,7 +869,65 @@ public async Task TestTraceMagic()
                 }
             ), 2);
         }
+
+        [TestMethod]
+        public async Task CompileAndSubmitWithClassicalControl()
+        {
+            const string source = @"
+                open Microsoft.Quantum.Measurement;
+
+                operation PrepareBellPair(left : Qubit, right : Qubit) : Unit is Adj + Ctl {
+                    H(left);
+                    CNOT(left, right);
+                }
+
+                operation Teleport(msg : Qubit, target : Qubit) : Unit {
+                    use register = Qubit();
+
+                    PrepareBellPair(register, target);
+                    Adjoint PrepareBellPair(msg, register);
+
+                    if MResetZ(msg) == One      { Z(target); }
+                    if MResetZ(register) == One { X(target); }
+                }
+
+                operation RunTeleport() : Unit {
+                    use left = Qubit();
+                    use right = Qubit();
+
+                    H(left);
+                    Teleport(left, right);
+                    H(right);
+
+                    if M(right) == One {
+                        fail ""Got wrong output from teleportation."";
+                    }
+                }
+            ";
+
+            var engineInput = await Assert.That
+                .UsingEngine()
+                .Input(source)
+                    .ExecutesSuccessfully()
+                .WithMockAzure()
+                .Input("%azure.target honeywell.mock")
+                    .ExecutesSuccessfully()
+                .Input("%azure.submit RunTeleport")
+                    .ExecutesSuccessfully()
+                .Input("%azure.target ionq.mock")
+                    .ExecutesSuccessfully()
+                .Input("%azure.submit RunTeleport")
+                    .ExecutesWithError(errors => errors.Any(error =>
+                        // Use StartsWith to avoid mismatching on newlines at
+                        // the end.
+                        error.StartsWith(
+                            "The Q# operation RunTeleport could not be " +
+                            "compiled as an entry point for job execution."
```

Comment:
I have to say the message here is kind of confusing.

Reply:
Fair enough; this was the message as it originally existed before this PR, such that this unit test only makes sure that the message is correctly omitted when submitting to targets that don't support the basic measurement feedback profile. As above, happy to consider changing the message, though.

Reply:
Source for the above error message: https://github.com/microsoft/iqsharp/blob/b789dcdeee32a578793f6be1a6db9f89c0c54161/src/AzureClient/AzureClient.cs#L318. If possible, for time constraints I'd prefer to defer an improvement to that message for now. Thank you!